### PR TITLE
feat(themes): add social share buttons to posts

### LINKED ISSUE #12: Social sharing links
## Summary

Add social sharing buttons to posts so readers can easily share content.

### Share Links
Generate share URLs for popular platforms:
- Twitter/X: `https://twitter.com/intent/tweet?url={url}&text={title}`
- LinkedIn: `https://www.linkedin.com/sharing/share-offsite/?url={url}`
- Facebook: `https://www.facebook.com/sharer/sharer.php?u={url}`
- Reddit: `https://reddit.com/submit?url={url}&title={title}`
- Hacker News: `https://news.ycombinator.com/submitlink?u={url}&t={title}`
- Email: `mailto:?subject={title}&body={url}`
- Copy link button (JS)

#### Option A: Template Helper
Jinja2 filter or macro:
```html
{% for platform, url in post | share_urls(site.url) %}
<a href="{{ url }}" target="_blank" rel="noopener">{{ platform }}</a>
{% endfor %}
```

#### Option B: Direct in Template
```html
<a href="https://twitter.com/intent/tweet?url={{ site.url }}{{ post.url }}&text={{ post.title | urlencode }}">
  Share on Twitter
</a>
```

### Styling
- Icon-only buttons (use SVG icons or icon font)
- Horizontal row at bottom of post
- Optional: floating sidebar on desktop
- Respect user's reduced-motion preferences

## Privacy Considerations

- Use direct share URLs (no tracking scripts)
- No embedded social widgets that track users
- Optional: add `rel="noopener noreferrer"` to links

### PR DESCRIPTION
**Pull request by x3ek**
Closes #12

Share buttons on post pages across all three bundled themes: Twitter/X, LinkedIn, Facebook, Reddit, Hacker News, Email, plus a copy-link button.

- Shared share_urls Jinja filter builds the links from the existing canonical_url with full percent-encoding (unicode titles included); returns empty when site.url is unset so the section hides entirely
- One shared partial (themes/default/_share.html) with inline SVG icons and a few lines of clipboard JS (navigator.clipboard with an execCommand fallback); blue-tech and terminal include it via the loader's default-theme fallback, zero duplication, content repos can override it
- Direct share URLs only, no tracking scripts or widgets; rel="noopener noreferrer" on external links; per-theme CSS matching each theme's look
- 16 tests: platform set and order, encoding edge cases, hidden case, and render tests across all three themes

Browser-verified on all three themes: share row renders in each theme's style, copy button shows its feedback state, links are correctly encoded, and the section is absent when site.url is not configured.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: themes/default/_share.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            ta.value = url;
+            ta.setAttribute("readonly", "");
+            ta.style.position = "absolute";
+            ta.style.left = "-9999px";
+            document.body.appendChild(ta);
+            ta.select();
+            document.execCommand("copy");
+            document.body.removeChild(ta);
+            done();
+        }
```
[the remark this answers is not in the document]
Comment (x3ek) [reply themes/default/_share.html:53]: Accepted in 2a7c452: writeText can exist and still reject (permissions policy, non-secure context edge cases), which left the button silent. The execCommand fallback is now also wired as the promise catch handler. Note for the diff view: an unrelated commit briefly landed on this branch by mistake and was reverted (2dc319b then 3142f7a); it never touched the share work and disappears in the squash.

*— Claude*
